### ISSUE TITLE
Build and run-time errors when using Ivy

### ISSUE THREAD

**Issue by 0-gravity** [telerik/kendo-angular#2321]
**Describe the bug**
Angular 8 fails to build a Grid when Ivy is enabled with error message: **ERROR in Cannot combine @Input decorators with query decorators**

**To Reproduce**

Used Angular cli - 8.0.2
1.) Create new project with flag: --enableIvy=true
2.) Add angular grid via: ng add @progress/kendo-angular-grid
3.) Import GridModule into AppModule
4.) Run command: ng serve

**Expected behavior**
Should pass a build

kendo-grid-compilation-failure@0.0.0 c:\Development\bugs\KendoGridCompilationFailure
+-- @angular-devkit/build-angular@0.800.2
+-- @angular/animations@8.0.0
+-- @angular/cli@8.0.2
+-- @angular/common@8.0.0
+-- @angular/compiler@8.0.0
+-- @angular/compiler-cli@8.0.0
+-- @angular/core@8.0.0
+-- @angular/forms@8.0.0
+-- @angular/language-service@8.0.0
+-- @angular/platform-browser@8.0.0
+-- @angular/platform-browser-dynamic@8.0.0
+-- @angular/router@8.0.0
+-- @progress/kendo-angular-buttons@4.4.2
+-- @progress/kendo-angular-dateinputs@3.7.4
+-- @progress/kendo-angular-dropdowns@3.5.6
+-- @progress/kendo-angular-excel-export@2.4.0
+-- @progress/kendo-angular-grid@3.14.4
+-- @progress/kendo-angular-inputs@5.2.2
+-- @progress/kendo-angular-intl@1.7.1
+-- @progress/kendo-angular-l10n@1.4.1
+-- @progress/kendo-angular-popup@2.6.0
+-- @progress/kendo-data-query@1.5.1
+-- @progress/kendo-drawing@1.5.12
+-- @progress/kendo-theme-default@3.6.1
+-- @types/jasmine@3.3.13
+-- @types/jasminewd2@2.0.6
+-- @types/node@8.9.5
+-- codelyzer@5.1.0
+-- jasmine-core@3.4.0
+-- jasmine-spec-reporter@4.2.1
+-- karma@4.1.0
+-- karma-chrome-launcher@2.2.0
+-- karma-coverage-istanbul-reporter@2.0.5
+-- karma-jasmine@2.0.1
+-- karma-jasmine-html-reporter@1.4.2
+-- protractor@5.4.2
+-- rxjs@6.4.0
+-- rxjs-compat@6.5.2
+-- ts-node@7.0.1
+-- tslib@1.10.0
+-- tslint@5.15.0
+-- typescript@3.4.5
`-- zone.js@0.9.1

**Comment by tsvetomir** [comment]
Support for Ivy is not in just yet. We'll follow-up when more information is available.

**Comment by markacx** [comment]
Any timeline for this

**Comment by tsvetomir** [comment]
Ivy is a bit of a moving target at the moment — some fixes we applied earlier are now obsolete. We can't commit to official support until things settle down a bit. We're currently working against angular@next 

Also, note that you should the `next` version of `@angular/core` when testing with Ivy, [see guide](https://angular.io/guide/ivy).

**Comment by tsvetomir** [comment]
This effort is currently blocked by https://github.com/angular/angular/issues/30080

**Comment by lormerod** [comment]
angular/angular#30080 is now closed.  Is kendo now working correctly under ivy?

**Comment by tsvetomir** [comment]
Not yet—we still have a few smaller issues to take care of. We're working together with the Angular team to resolve these. They've set up a project dedicated to validating backwards-compatibility with Ivy, see https://github.com/angular/ngcc-validation/issues/386 for more details.

**Comment by tsvetomir** [comment]
Updating the issue title as Ivy support affects all components, not just the Grid.

**Comment by thaoula** [comment]
@tsvetomir so would you want us to log ivy build issues like the following -

DtsModuleScopeResolver.read(LocalizedMessagesDirective$3 from /Users/thaoula/Projects/Empower/jms-platform/jms-web/node_modules/@progress/kendo-angular-dateinputs/dist/fesm2015/index.js), but not a .d.ts file

Or should we wait until you guys finish with the angular team?

**Comment by tsvetomir** [comment]
@thaoula sorry for the delay!

This issue has been fixed in the development builds of the Date Inputs. Update by running `npm install --save @progress/kendo-angular-dateinputs@dev` or `yarn add @progress/kendo-angular-dateinputs@dev`.

Both comments and separate issues are fine. I hope most people subscribed to this thread will find the early warnings useful.

**Comment by servelec-jack-starkey** [comment]
Hi, I'm having similar issues while trying to build in ivy mode with a grid:
`ERROR in ../../node_modules/@progress/kendo-angular-grid/dist/es2015/index.d.ts:101:9 - error TS2304: Cannot find name 'PDFMarginComponent'.
101 export {PDFMarginComponent as PDFMarginComponent$1};
            ~~~~~~~~~~~~~~~~~~
../../node_modules/@progress/kendo-angular-grid/dist/es2015/index.d.ts:102:9 - error TS2304: Cannot find name 'PDFTemplateDirective'.
102 export {PDFTemplateDirective as PDFTemplateDirective$1};
            ~~~~~~~~~~~~~~~~~~~~
../../node_modules/@progress/kendo-angular-grid/dist/es2015/pdf/pdf.module.d.ts:38:25 - error TS7016: Could not find a declaration file for module '../../fesm2015/index'. '/home/[repo]/node_modules/@progress/kendo-angular-grid/dist/fesm2015/index.js' implicitly has an 'any' type.
38 import * as ɵngcc2 from '../../fesm2015/index';`

`  ERROR in node_modules/@progress/kendo-angular-grid/dist/es2015/index.d.ts(101,9): error TS2304: Cannot find name 'PDFMarginComponent'.
    node_modules/@progress/kendo-angular-grid/dist/es2015/index.d.ts(102,9): error TS2304: Cannot find name 'PDFTemplateDirective'.
    node_modules/@progress/kendo-angular-grid/dist/es2015/pdf/pdf.module.d.ts(38,25): error TS7016: Could not find a declaration file for module '../../fesm2015/index'. '/home/[repo]/node_modules/@progress/kendo-angular-grid/dist/fesm2015/index.js' implicitly has an 'any' type.`


Anything you can advise that I do to get it to compile? Thanks!

**Comment by tomvanoost** [comment]
@servelec-jack-starkey Same situation here. I get these errors on the latest development build.

**Comment by tsvetomir** [comment]
We don't have a workaround for that particular issue yet. It looks like a backwards-compatibility issue with ngcc thay we hope to get solved in the upcoming Angular RCs.

**Comment by servelec-jack-starkey** [comment]
Hi @tsvetomir do you have a link to the issue in the Angular repo?

**Comment by tsvetomir** [comment]
It's not in the public issue tracker AFAIK, see the [ngcc failures](https://docs.google.com/spreadsheets/d/1brJHrqGYGGD4HRsFudGW6CynHHX34Zf8H31qD7wjPac/edit#gid=0) summary document.

**Comment by servelec-jack-starkey** [comment]
> It's not in the public issue tracker AFAIK, see the [ngcc failures](https://docs.google.com/spreadsheets/d/1brJHrqGYGGD4HRsFudGW6CynHHX34Zf8H31qD7wjPac/edit#gid=0) summary document.

Thanks @tsvetomir, I may be reading that incorrectly but the colour coding indicates it is a library (Kendo) issue?

**Comment by tsvetomir** [comment]
I don't have enough information at the moment, but it may very well be an issue with our code.

**Comment by gradx** [comment]
This is an issue for Angular 9.0.0-rc.5 as well

**Comment by tsvetomir** [comment]
Compilation issues are fixed in [Angular 9.0.0-rc.7](https://github.com/angular/angular/blob/master/CHANGELOG.md#900-rc7-2019-12-18) through angular/angular#34254.

**Comment by tsvetomir** [comment]
Closing this issue as the compilation errors are now solved for all components. That is not to say that the components are fully compatible with Ivy just yet—we'll validate this in the upcoming days.

**Comment by bwinchester** [comment]
Just upgraded to 9.0.0-rc7-next to get this to work, seems to still be broken...reallly dont like paying for this.  
<img width="761" alt="image" src="https://user-images.githubusercontent.com/2374757/71380065-84ffd380-2593-11ea-902c-34bb4c4bf8b0.png">

**Comment by tsvetomir** [comment]
@bwinchester can you please file a new issue with more details about the error you're getting? Thanks!

**Comment by 0-gravity** [comment]
Project from scratch gets compiled with RC7.

**Comment by 0-gravity** [comment]
> Just upgraded to 9.0.0-rc7-next to get this to work, seems to still be broken...reallly dont like paying for this.
> <img alt="image" width="761" src="https://user-images.githubusercontent.com/2374757/71380065-84ffd380-2593-11ea-902c-34bb4c4bf8b0.png">

Have you tried to upgrade components version?

**Comment by bwinchester** [comment]
Issue was Ivy was still set to true for compile... all my fault.

> > Just upgraded to 9.0.0-rc7-next to get this to work, seems to still be broken...reallly dont like paying for this.
> > <img alt="image" width="761" src="https://user-images.githubusercontent.com/2374757/71380065-84ffd380-2593-11ea-902c-34bb4c4bf8b0.png">
> 
> Have you tried to upgrade components version?

**Comment by MaklaCof** [comment]
@tsvetomir  In [this comment](https://github.com/telerik/kendo-angular/issues/2321#issuecomment-567377628), you said:

> Closing this issue as the compilation errors are now solved for all components. That is not to say that the components are fully compatible with Ivy just yet—we'll validate this in the upcoming days.

Is there anything new about official support for Angular 9 for all components?

**Comment by tsvetomir** [comment]
The components are compatible with the latest release candidates with some minor exceptions. We're currently tracking a few issues with the Sortable component that we hope to fix soon.

**Comment by tscislo** [comment]
@tsvetomir does that mean that I need to bump all kendo components to latest version and it should be fine to compile with Ivy and Angular 9.0.0?

**Comment by tsvetomir** [comment]
@tscislo yes, all [latest versions](https://www.telerik.com/kendo-angular-ui/components/installation/up-to-date/#toc-updating-to-latest-versions) of the packages are compatible with Angular 9.
